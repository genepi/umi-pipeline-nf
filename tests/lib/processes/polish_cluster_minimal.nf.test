nextflow_process {

    name "Test Process POLISH_CLUSTER_minimal"
    script "lib/processes/polish_cluster.nf"
    process "POLISH_CLUSTER"

    test("Should use gpu") {
        when {
            params{
                use_gpu = true
            }
            process {
                """
                input[0] = Channel.of(["barcode03", "raw", "$baseDir/tests/input/polish_cluster/barcode03/smolecule0sub12.fastq"])
                input[1] = "Consensus"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
