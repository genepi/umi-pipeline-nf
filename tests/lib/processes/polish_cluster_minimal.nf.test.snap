{
    "Should use gpu": {
        "content": [
            {
                "0": [
                    [
                        "barcode03",
                        "raw",
                        "smolecule0sub12_consensus.fastq:md5,f284c702025b387dcf4bc2c3aa8af37b"
                    ]
                ],
                "consensus_fastq": [
                    [
                        "barcode03",
                        "raw",
                        "smolecule0sub12_consensus.fastq:md5,f284c702025b387dcf4bc2c3aa8af37b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "22.10.3"
        },
        "timestamp": "2024-09-13T12:32:01.541104"
    },
    "Should use gpu and singularity": {
        "content": [
            {
                "0": [
                    [
                        "barcode03",
                        "raw",
                        "smolecule0sub12_consensus.fastq:md5,f284c702025b387dcf4bc2c3aa8af37b"
                    ]
                ],
                "consensus_fastq": [
                    [
                        "barcode03",
                        "raw",
                        "smolecule0sub12_consensus.fastq:md5,f284c702025b387dcf4bc2c3aa8af37b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "22.10.3"
        },
        "timestamp": "2024-09-13T14:27:10.003854"
    }
}